thrust/system/cuda/detail/bulk/malloc.hpp

```diff
@@ -38,9 +38,24 @@ inline __device__ bool is_on_chip(void *ptr)
 template<typename T>
 inline __device__ T *on_chip_cast(T *ptr)
 {
+#if defined(__CUDA__) && defined(__NVCC__) && !defined(__clang__)
```

Comment:
Yikes, sorry about that.  I checked the capitalization of `__NVCC__` but it didn't even occur to me that nvcc wouldn't define `__CUDA__`.

> We need something like

    #ifdef __NVCC__  

except this doesn't work because clang defines `__NVCC__` (which I'm not at all happy about, but that's another story)

> or

    #if !(defined(__CUDA__) && defined(__clang__))

except we said we wanted this to trigger only if nvcc, and this triggers if not clang.

So I think we may *actually* want to detect "really NVCC" with something like

    #if defined(__NVCC__) && !(defined(__CUDA__) && defined(__clang__))

I'll send a patch if we agree this is right?